pkg/apis/settings/install/install.go

```diff
@@ -22,17 +22,15 @@ import (
 	"k8s.io/apimachinery/pkg/apimachinery/announced"
 	"k8s.io/apimachinery/pkg/apimachinery/registered"
 	"k8s.io/apimachinery/pkg/runtime"
-	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/apis/settings"
 	"k8s.io/kubernetes/pkg/apis/settings/v1alpha1"
 )
 
-func init() {
-	Install(api.GroupFactoryRegistry, api.Registry, api.Scheme)
-}
-
 // Install registers the API group and adds types to a scheme
 func Install(groupFactoryRegistry announced.APIGroupFactoryRegistry, registry *registered.APIRegistrationManager, scheme *runtime.Scheme) {
+	if _, ok := groupFactoryRegistry[settings.GroupName]; ok {
```

Comment:
@lavalamp @sttts i reverted to this commit.

At this version, install.go returns early if the group is already announced. I think this is slightly better than adding an `AnnounceIfNotAnnounced` function, because it keeps the `announced` package clean.

This PR is not making the multi-registration problem worse, people are already doing multiple-registration and we were relying on `init()` to hide that fact. Completely removing multi-registration is a too big refactor than i can handle right now.

Please let me know if you think the PR is good enough.